CONTRIBUTING.md

```diff
@@ -1,5 +1,17 @@
 # Contributing
 
+## Quick Start
+
+    $ git clone git@github.com:puppetlabs/bolt.git
+    $ cd bolt
+    $ git submodule init
+    $ git submodule update
+    $ bundle install --path .bundle/gems/
+    $ bundle exec rake unit
```

Comment:
This is partial covered in `INSTALL.md` and mentioned in the [`README.md`](https://github.com/puppetlabs/bolt/blob/master/README.md#bolt-raises-the-error-puppet-must-be-installed-to-execute-tasks) also. That said having a more developer-focused walk-through isn't a bad idea. Also nit, we've been removing prompts `$` from code examples to make it easier to copy & paste.

Reply:
Also good to encourage people to use vagrant and run the full tests with `rake test` if possible

Reply:
I would agree that running the full vagrant test is nice but having both how to run the unit tests and pull down the submodules would be great as I got bit by this the other day until I talked to Alex Dreyer to see the submodule docs.